warehouse/legacy/api/json.py

```diff
@@ -217,3 +217,120 @@ def json_release_slash(release, request):
         ),
         headers=_CORS_HEADERS,
     )
+
+
+@view_config(
+    route_name="legacy.api.json.latest",
+    context=Project,
+    renderer="json",
+    decorator=_CACHE_DECORATOR,
+)
+def json_latest(project, request):
+    release = project.latest_version
+    version = release.version if release else None
+
+    if version is None:
+        return HTTPNotFound(headers=_CORS_HEADERS)
```

Comment:
Aswering https://github.com/pypa/warehouse/pull/8615/files#r618602220

We could do
```suggestion
    if release is None:
        return HTTPNotFound(headers=_CORS_HEADERS)
        
    version = release.version
```
I believe a release MUST have a non-None version.

Reply:
Implemented along these lines in 688d842e. Also removed redundant database lookups.